### PR TITLE
Sharing an OpenGL context between level rendering and 2d SDL rendering

### PR DESCRIPTION
**Pull request by Interrupt**

### CODE REVIEW

**Comment by Interrupt** [comment]
This simplifies things down to just use one OpenGL context instead of two, to fix a bunch of OSX related issues.

**Comment by inguin** [comment]
Looks good in Linux! The OpenGL renderer now always fills the entire window area instead of preserving aspect ratio, but that may be intended.

It does not seem to work in Windows, though. This is how the game looks built with mingw32 and running in wine:

screenshot_20180811_111136

File: src/GameSrc/fullscrn.c [hunk 1]
```diff
@@ -53,6 +53,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "tools.h"
 #include "wares.h"
 
+#include "OpenGl.h"
```
Comment (inguin) [inline src/GameSrc/fullscrn.c:56]: Wrong case, should be "OpenGL.h" with a capital L.

File: src/MacSrc/OpenGL.cc [hunk 2]
```diff
@@ -228,6 +228,10 @@ static void set_blend_mode(bool enabled) {
             glDisable(GL_BLEND);
         }
     }
+
+    glDisable(GL_BLEND);
+    glAlphaFunc(GL_GEQUAL, 1.0f);
```
Comment (inguin) [inline src/MacSrc/OpenGL.cc]: This looks wrong. With this change, everything except the palette maximum-brightness pal-fx pixels is black.

**Comment by Interrupt** [comment]
SDL_Renderer on Windows must not be using OpenGL ES from the looks of it, going to check and see if there's anything else I can do to reset the render state at the end of the frame for that renderer.

**Comment by inguin** [comment]
I'm not really convinced this is a good idea in the first place. Using separate contexts sounds more future-proof than trying to guess what state the SDL renderer might clobber or depend on. What was the problem with that approach on OS-X?

**Comment by Interrupt** [comment]
It caused a ton of issues - I was having to do some really weird hacks to get the right contexts bound otherwise weird stuff would happen. Usually the UI was rendering interleaved under parts of the world. 

The more I'm looking into this, the more I'm thinking it would be a good idea to ditch SDL Renderer entirely eventually and just manage drawing ourselves, it would't be that much more difficult. That's the only foolproof way to do this - especially if we want multiple rendering backends besides OpenGL.

**Comment by inguin** [comment]
Strange! In theory there should be no more than two context switches per frame. Was it always that weird or did it start with a specific commit?

**Comment by Interrupt** [comment]
It was always like that, I found a hacky fix for it in this commit: https://github.com/inguin/systemshock/pull/11/commits/3f0df412f1761b78ab43ab29868e11d715d4081f

For some reason generating a new texture would force some context things to happen where I wanted them to. Why? No idea :D

**Comment by inguin** [comment]
Have you tried adding a `glFlush()` at this place? Not sure when and where this is needed.

**Comment by Interrupt** [comment]
I haven't, that's probably a good thing to try there.

**Comment by Interrupt** [comment]
Going to close this one, opened a new simpler PR.

**Pull request closed without merging**